rotkehlchen/tests/api/test_addressbook.py

```diff
@@ -281,40 +384,123 @@ def names_request(addresses: list[str]):
+        AddressbookEntry(address=address_tylor, blockchain=SupportedBlockchain.ETHEREUM, name='Tylor'),  # noqa: E501
+        AddressbookEntry(address=address_cody, blockchain=SupportedBlockchain.ETHEREUM, name='Cody'),  # noqa: E501
+        AddressbookEntry(address=address_rose, blockchain=SupportedBlockchain.ETHEREUM, name='Rose'),  # noqa: E501
     }
     response = names_request(private_addressbook_addresses)
     result = assert_proper_response_with_result(response)
-    assert result == private_addressbook_expected
+    assert {AddressbookEntry.deserialize(x) for x in result} == private_addressbook_expected
+
+
+@pytest.mark.parametrize('book_type', [AddressbookType.GLOBAL, AddressbookType.PRIVATE])
+def test_insert_into_addressbook_no_blockchain(
+        rotkehlchen_api_server: 'APIServer',
+        book_type: AddressbookType,
+) -> None:
+    """
+    Check that adding a name with no blockchain works as expected and it replaces any other
+    entry if it exists.
+    """
+    test_address = to_checksum_address('0x9531c059098e3d194ff87febb587ab07b30b1306')
+
+    # add a name for blockchain ethereum
+    custom_name = AddressbookEntry(
+        address=test_address,
+        name='my address',
+        blockchain=SupportedBlockchain.ETHEREUM,
+    )
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'entries': [custom_name.serialize()],
+        },
+    )
+    assert_proper_response_with_result(response)
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'addresses': [
+                {'address': test_address},
+            ],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    assert result == [custom_name.serialize()]
+
+    # now add the same name for all blockchains and see the value replaced
+    custom_name = AddressbookEntry(
+        address=test_address,
+        name='my address',
+        blockchain=None,
+    )
+    response = requests.put(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'entries': [custom_name.serialize()],
+        },
+    )
+    assert_proper_response_with_result(response)
+    response = requests.post(
+        api_url_for(
+            rotkehlchen_api_server,
+            'addressbookresource',
+            book_type=book_type,
+        ),
+        json={
+            'addresses': [
+                {'address': test_address},
+            ],
+        },
+    )
+    result = assert_proper_response_with_result(response)
+    assert result == [custom_name.serialize()]
```

Comment:
Shouldn't you also check the DB in this test? Or you can just move that to a unit test of the corresponding DB function.

What's more ... do we have a test for addign same name but for different chain? For example:

1. Add a name for 0xyabir in ethereum.
2. Add same name for 0xyabir in optimism.

What happens then? Do we have a test for this? I guess what "should" happen is API error.

Reply:
Why an error? Having same or different name for address `0xyabir` on different chains is allowed. Otherwise the primary key would be name and address and not blockchain and address as it is now

Reply:
Yes okay true. But can you add a test for this? To see same name for different addresses across different chains works?

But then also what happens if:

1. I add "yabiroptimism" for "0x1" in optimism
2. I add "yabirethereum" for "0x1" in ethereum.
3. Add "yabir" for "0x1" for chain=None?

Reply:
We had them in the inser. In the case where we insert for chain = None I have added it here `test_insert_into_addressbook`. As we agreed it deletes the other two entries and leaves the one with `chain=None`